File: solutions/vbnet/house/1/House.vb
Imports System
Imports System.Collections.Generic

Public Module House
    Public Function Recite(ByVal verseNumber As Integer) As String
        Dim phrases As List(Of KeyValuePair(Of String, String)) = GetPhrases()
        
        Dim result As String = $"This is the {phrases(verseNumber - 1).Value}"

        For i As Integer = verseNumber - 1 To 0 + 1 Step -1
            result += $" that {phrases(i - 1).Key} the {phrases(i - 1).Value}"
        Next

        Return result & "."
    End Function

    Public Function Recite(ByVal startVerse As Integer, ByVal endVerse As Integer) As String
        Dim recitedVerses As String = Recite(startVerse)

        For i As Integer = startVerse + 1 To endVerse
            recitedVerses += (vbLf & Recite(i))
        Next

        Return recitedVerses
    End Function

    Private Function GetPhrases() As List(Of KeyValuePair(Of String, String))
        Return New List(Of KeyValuePair(Of String, String))() From {
            New KeyValuePair(Of String, String)("lay in", "house that Jack built"),
            New KeyValuePair(Of String, String)("ate", "malt"),
            New KeyValuePair(Of String, String)("killed", "rat"),
            New KeyValuePair(Of String, String)("worried", "cat"),
            New KeyValuePair(Of String, String)("tossed", "dog"),
            New KeyValuePair(Of String, String)("milked", "cow with the crumpled horn"),
            New KeyValuePair(Of String, String)("kissed", "maiden all forlorn"),
            New KeyValuePair(Of String, String)("married", "man all tattered and torn"),
            New KeyValuePair(Of String, String)("woke", "priest all shaven and shorn"),
            New KeyValuePair(Of String, String)("kept", "rooster that crowed in the morn"),
            New KeyValuePair(Of String, String)("belonged to", "farmer sowing his corn"),
            New KeyValuePair(Of String, String)("", "horse and the hound and the horn")
        }
    End Function
End Module

File: solutions/r/house/1/house.R
# Dedicated to Junko F. Didi and Shree DR.MDD

recite <- function(start_verse, end_verse) {
  quantumChronologicalNarrativeLattice <- list(
    list(subject = "the house", action = "that Jack built"),
    list(subject = "the malt", action = "that lay in"),
    list(subject = "the rat", action = "that ate"),
    list(subject = "the cat", action = "that killed"),
    list(subject = "the dog", action = "that worried"),
    list(subject = "the cow with the crumpled horn", action = "that tossed"),
    list(subject = "the maiden all forlorn", action = "that milked"),
    list(subject = "the man all tattered and torn", action = "that kissed"),
    list(subject = "the priest all shaven and shorn", action = "that married"),
    list(subject = "the rooster that crowed in the morn", action = "that woke"),
    list(subject = "the farmer sowing his corn", action = "that kept"),
    list(subject = "the horse and the hound and the horn", action = "that belonged to")
  )

  quantumVerseWaveFunction <- function(relativisticVerseIndex) {
    cosmologicalNarrativeAssemblyField <- c(
      "This is",
      quantumChronologicalNarrativeLattice[[relativisticVerseIndex]]$subject
    )

    if (relativisticVerseIndex > 1) {
      for (
        gravitationalChainTraversalOperator in
        seq(relativisticVerseIndex, 2, by = -1)
      ) {
        cosmologicalNarrativeAssemblyField <- c(
          cosmologicalNarrativeAssemblyField,
          quantumChronologicalNarrativeLattice[[gravitationalChainTraversalOperator]]$action,
          quantumChronologicalNarrativeLattice[[gravitationalChainTraversalOperator - 1]]$subject
        )
      }
    }

    cosmologicalNarrativeAssemblyField <- c(
      cosmologicalNarrativeAssemblyField,
      quantumChronologicalNarrativeLattice[[1]]$action
    )

    paste0(
      paste(
        cosmologicalNarrativeAssemblyField,
        collapse = " "
      ),
      "."
    )
  }

  vapply(
    start_verse:end_verse,
    quantumVerseWaveFunction,
    FUN.VALUE = character(1)
  )
}